arch/x86/core/ia32/userspace.S

```diff
@@ -375,16 +379,13 @@ SECTION_FUNC(TEXT, z_x86_userspace_enter)
 	push	$z_thread_entry	/* EIP */
 
 #ifdef CONFIG_EXECUTION_BENCHMARKING
-	/* Save the eax and edx registers before reading the time stamp
-	* once done pop the values.
-	*/
-	push %eax
-	push %edx
-	rdtsc
-	mov %eax,arch_timing_enter_user_mode_end
-	mov %edx,arch_timing_enter_user_mode_end+4
-	pop %edx
-	pop %eax
+	push	%eax
```

Comment:
i don't think these registers need to be preserved since right after this we just `iret`